src/foam/core/reflow/dashboard/DashboardDAOAgents.js

```diff
@@ -0,0 +1,1218 @@
+          borderWidth: 2,
+          fill: false,
+          tension: 0.1
+        }]
+      };
+    },
+    
+    function addToE(e) {
+      e.startContext({data: this}).
+        start().
+          style({display: 'flex', gap: '10px', flexWrap: 'wrap'}).
+          add('X Property: ', this.X_PROP).
+          add('Y Property: ', this.Y_PROP).
+          add('Time Unit: ', this.TIME_UNIT).
+        end().
+      endContext();
+    }
+  ]
+});
+
+foam.CLASS({
+  package: 'foam.core.reflow.dashboard',
+  name: 'DashboardMetricDAOAgent',
+  extends: 'foam.core.reflow.AbstractSinkDAOAgent',
+  mixins: [
+    'foam.core.reflow.dashboard.CardRenderMixin'
+  ],
+
+  requires: [
+    'foam.mlang.sink.Count',
+    'foam.mlang.sink.Sum',
+    'foam.mlang.sink.Min',
+    'foam.mlang.sink.Max',
+    'foam.mlang.sink.Average',
+    'foam.core.reflow.dashboard.MetricOperation'
+  ],
+
+  properties: [
+    {
+      class: 'Enum',
+      of: 'foam.core.reflow.dashboard.MetricOperation',
+      name: 'operation',
+      value: 'COUNT'
+    },
+    {
+      name: 'prop',
+      view: function(_, X) {
+        return { 
+          class: 'foam.core.reflow.PropertyChoiceView', 
+          forCls: X.data.dao.of
+        };
+      }
+    },
+    {
+      class: 'String',
+      name: 'label',
+      value: 'Metric'
+    }
+  ],
+
+  methods: [
+    function execute(e) {
+      var self = this;
+      
+      if (this.operation !== 'COUNT' && !this.prop) {
+        this.showPropertyRequiredMessage(e);
+        return;
+      }
+
+      var sink = this.operation.createSink(this.prop);
+      this.dao.select(sink).then(function(result) {
+        // Render using CardRenderMixin
+        self.renderMetricValue(e, self.getDisplayLabel(), result.value);
+
+        if (self.block) {
+          if (self.block.value) {
+            self.block.value.value = result.value;
+          } else {
+            self.block.value = result.value;
+          }
+        }
+      });
+    },
+
+    function getDisplayLabel() {
+      return this.label || 
+             (this.prop ? this.operation.label + ' of ' + this.prop.label : this.operation.label);
+    },
+
+    function showPropertyRequiredMessage(e) {
+      e.start('div').
+        style({padding: '20px', textAlign: 'center', color: foam.CSS.returnTokenValue('$textTertiary', this.cls_, this.__context__)}).
+        add('Please select a property for ' + this.operation.label + ' operation').
+      end();
+    },
+
+    function addToE(e) {
+      e.startContext({data: this}).
+        start().
+          style({display: 'flex', gap: '10px', flexWrap: 'wrap'}).
+          add('Operation: ', this.OPERATION).
+          add('Property: ', this.PROP).
+          add('Label: ', this.LABEL).
+          add('Render as Card: ', this.RENDER_AS_CARD).
+        end().
+      endContext();
+    }
+  ]
+});
+
+foam.CLASS({
+  package: 'foam.core.reflow.dashboard',
+  name: 'DashboardGridDAOAgent',
```

Comment:
Shouldnt be needed anymore with `layout` command

Reply:
ok, will remove